### PR TITLE
Fix some Codacy warnings

### PR DESCRIPTION
**Pull request by MarcelRaad**
Mostly remove redundant code and reduce variable scopes.

### CODE REVIEW

File: lib/ftp.c [hunk 1]
```diff
@@ -2482,9 +2482,9 @@ static CURLcode ftp_state_get_resp(struct Curl_easy *data,
           bytes--;
         }
         /* if we have nothing but digits: */
-        if(bytes++) {
+        if(bytes) {
           /* get the number! */
-          (void)curlx_strtoofft(bytes, NULL, 0, &size);
+          (void)curlx_strtoofft(++bytes, NULL, 0, &size);
```
Comment (danielgustafsson) [inline lib/ftp.c:2487]: Minor nitpick: wouldn't it be more readable to just do `bytes++;` on its own line before the  `curlx_strtoofft` call?

Comment (MarcelRaad) [reply lib/ftp.c:2487]: Fine for me too. Personally, I find both equally readable. What do others think?

Comment (jzakrzewski) [reply lib/ftp.c:2487]: As much as I like terse code, `++bytes;` on a separate line makes it easier not to miss. It's not a strong opinion, however.

Comment (bagder) [reply lib/ftp.c:2487]: I'm equally fine with either.

**Review by bagder — APPROVED** [review]

File: lib/ftp.c (hunk 1 above)
Comment (MarcelRaad) [reply lib/ftp.c:2487]: I put it on a separate line in the final commit that landed on master.

**Pull request closed without merging**